go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -1000,13 +1000,38 @@ func (root *RootValue) ValidateForeignKeysOnSchemas(ctx context.Context) (*RootV
 	for _, foreignKey := range allForeignKeys {
 		tblSch, existsInRoot := allTablesSet[foreignKey.TableName]
 		if existsInRoot {
+			// Check that the foreign key table has a valid schema
 			if err := foreignKey.ValidateTableSchema(tblSch); err != nil {
 				return nil, err
 			}
 			parentSch, existsInRoot := allTablesSet[foreignKey.ReferencedTableName]
 			if !existsInRoot {
 				return nil, fmt.Errorf("foreign key `%s` requires the referenced table `%s`", foreignKey.Name, foreignKey.ReferencedTableName)
 			}
+
+			// Check if the foreign key needs to be resolved
+			if foreignKey.HasDelayedResolution() {
```

Comment:
We don't do any validation if foreign key checks are disabled right? We need that check in this function.